src/scheduler/node_info.c

```diff
@@ -5093,21 +5091,6 @@ reorder_nodes(node_info **nodes, resource_resv *resresv)
 		snprintf(last_node_name, sizeof(last_node_name), "%s", nodes[0]->name);
 
 	if (resresv != NULL) {
-		if (resresv->is_resv && resresv->resv != NULL && resresv->resv->check_alternate_nodes) {
-			int		i = 0;
-			node_info	*temp = NULL;
-
-			memcpy(nptr, nodes, (nsize + 1) * sizeof(node_info *));
-			for (i = 0; nptr[i] != NULL; i++) {
-				temp = find_node_by_rank(resresv->ninfo_arr, nptr[i]->rank);
-				if (temp != NULL)
-					nptr[i]->nscr.to_be_sorted = 0;
-				else
-					nptr[i]->nscr.to_be_sorted = 1;
-			}
-			qsort(nptr, i, sizeof(node_info*), cmp_nodes_sort);
```

Comment:
You might want to remove cmp_nodes_sort as well. looks like it is of no use now.